### PR TITLE
Bump HeatpumpIR and IRremoteESP8266

### PR DESCRIPTION
**Pull request by nagyrobi**
IRremoteESP8266 to 2.8.6: https://github.com/crankyoldgit/IRremoteESP8266/releases/tag/v2.8.6
HeatpumpIR to 1.0.26: https://github.com/ToniA/arduino-heatpumpir/commits/1.0.26

Add new protocols:
```
greeyap
nibe
carrier_qlima_1
carrier_qlima_2
samsung_aqv12msan
zhjg01
```

# What does this implement/fix?

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#3975

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [x] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by nagyrobi** [comment]
Tested, working with
```yaml
climate:
  - platform: heatpumpir
    id: ac_room
    name: A/C room
    protocol: greeyan
    sensor: ha_temperature
    horizontal_default: auto
    vertical_default: auto
    min_temperature: 18.0
    max_temperature: 30.0
    visual:
      temperature_step:
        current_temperature: 0.1
        target_temperature: 1.0
```

**Review by jesserockz — DISMISSED** [review]

**Comment by nagyrobi** [comment]
I just noticed by grepping into the libary's *.cpp files that it supports more than what's included by this component, specifically these are missing:
```
AIRWAY
BGH_aud
panasonic_altdke
qlima_1
qlima_2
r51m
samsung_aqv12msan
vaillantvai8
ZHJG01
```
Lemme add these too

**Comment by nagyrobi** [comment]
Unfortunately
```
airway
bgh_aud
panasonic_altdke
vaillantvai8
r51m
```
could not be added because arduino-heatpumpir/HeatpumpIRFactory.h is missing inclusion of their corresponding header files.
PR made at https://github.com/ToniA/arduino-heatpumpir/pull/174. 
Until that gets merged, I commented out the corresponding entries on our side. As soon as they merge the PR above, all that needs to be done is to uncomment these lines here.

This can be merged meanwhile, I'll remove the comments in another PR (the heatpumpir repo moves slower apparently).

File: esphome/components/heatpumpir/climate.py [hunk 1]
```diff
@@ -61,6 +62,16 @@
...
+    "nibe": Protocol.PROTOCOL_NIBE,
+    "carrier_qlima_1": Protocol.PROTOCOL_QLIMA_1,
+    "carrier_qlima_2": Protocol.PROTOCOL_QLIMA_2,
+    "samsung_aqv12msan": Protocol.PROTOCOL_SAMSUNG_AQV12MSAN,
+    "zhjg01": Protocol.PROTOCOL_ZHJG01,
+    # "airway": Protocol.PROTOCOL_AIRWAY,
+    # "bgh_aud": Protocol.PROTOCOL_BGH_AUD,
+    # "panasonic_altdke": Protocol.PROTOCOL_PANASONIC_ALTDKE,
+    # "vaillantvai8": Protocol.PROTOCOL_VAILLANTVAI8,
+    # "r51m": Protocol.PROTOCOL_R51M,
```
Comment (jesserockz) [inline esphome/components/heatpumpir/climate.py]: I can't really let you add in commented out code if we don't let other people do it...

Comment (nagyrobi) [reply esphome/components/heatpumpir/climate.py]: Ok... I'll save it in a private .txt for later.
Tomorrow....

**Comment by jesserockz** [comment]
Please also add a link to the library changelogs/releases in the PR description.
Makes it easier to do a quick check of external code to make sure we are not going to bring something malicious in.

File: esphome/components/heatpumpir/climate.py (hunk 1 above)
Comment (nagyrobi) [reply esphome/components/heatpumpir/climate.py]: Done

**Review by jesserockz — APPROVED** [review]

**Comment by nagyrobi** [comment]
> HeatpumpIRFactory.h

@rob-deutsch Do you think we could eliminate dependency on HeatpumpIRFactory.h on our side? As it appears at heappumpir they either forget updating it, or leaving out some protocols maybe for a reason?

**Comment by rob-deutsch** [comment]
Before I/we answer that question I think we should ask the ToniA (the maintainer of the `arduino-heatpumpir` library) - how do they want their library to be used?

Do they want the `HeatpumpIRFactory.h` to be a list of "approved protcols"? Or would they rather users refer to the individual protocol headers directly?

I'll ask the question in https://github.com/ToniA/arduino-heatpumpir/pull/174

**Pull request merged**